avm/res/operational-insights/cluster/tests/e2e/cmk-hsm-uami/main.test.bicep

```diff
@@ -0,0 +1,94 @@
+targetScope = 'subscription'
+
+metadata name = 'Using managed HSM Customer-Managed-Keys with User-Assigned identity'
+metadata description = 'This instance deploys the module with Managed HSM-based Customer Managed Key (CMK) encryption, using a User-Assigned Managed Identity to access the HSM key.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-operational-insights-cluster-${serviceShort}-rg'
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'oichsmu'
+
+@description('Optional. A token to inject into the name of each resource. This value can be automatically injected by the CI.')
+param namePrefix string = '#_namePrefix_#'
+
+@description('Generated. Used as a basis for unique resource names.')
+param baseTime string = utcNow('u')
+
+@description('Required. The resource ID of the Managed Identity used by the deployment script. This value is tenant-specific and must be stored in the CI Key Vault in a secret named \'CI-deploymentMSIName\'.')
+@secure()
+param deploymentMSIResourceId string = ''
+
+@description('Required. The resource ID of the managed HSM used for encryption. This value is tenant-specific and must be stored in the CI Key Vault in a secret named \'CI-managedHSMResourceId\'.')
+@secure()
+param managedHSMResourceId string = ''
+
+// Enforce location of HSM
+var enforcedLocation = 'uksouth'
```

Comment:
Should we enforce northeurope to track that that is the region which worked? As far as I understood uksouth would fail

Reply:
well we know for sure that it will fail :D

Reply:
If we run it right now, yes - due to (temporal) capacity issue. If I change it to NorthEurope it will always, not matter what, as we don't have an HSM there. Not sure if that's better 😄